liquidprompt

```diff
@@ -2971,6 +2994,48 @@ _lp_git_staged_lines() {
     lp_vcs_staged_d_lines=$lp_git_diff_shortstat_d_lines
 }
 
+# Get the remote name for the current branch.
+# NOTE: Assumes that `_lp_find_vcs` and `_lp_vcs_branch` (costly functions) have been called.
+_lp_git_remote() {
+    lp_vcs_remote=""
+
+    # If there is no branch…
+    if [[ -z "$lp_vcs_branch" ]]; then
+        # … then there is no remote.
+        return 1
+    fi
+
+    # If we don't have a VCS root…
+    if [[ -z "$lp_vcs_root" ]]; then
+        # … we cannot find the config file.
+        return 1
+    fi
+
+    local file="${lp_vcs_root}/.git/config"
+    local delimiter=" = " # With spaces.
+    local in_branch=0
+    # We test for $line in the loop is here to ensure that we read the last line,
+    # even if the file does not ends with a \n.
+    # This bypass a known behavior of the C standard, not fixed in POSIX.
+    while IFS='' read -r line || [[ -n "$line" ]] ; do
+        # If we are in the branch config section.
+        if [[ "$line" == "[branch \"${lp_vcs_branch}\"]" ]]; then
+            in_branch=1
+        elif [[ "$line" == *"remote${delimiter}"* ]] ; then
+            if [[ "$in_branch" ]]; then
```

Comment:
Apparently this is working, but I didn't know that this syntax worked for variables with either `1` or `0`.

Reply:
I should have trusted my gut. This is not correct syntax. `[[ $var ]]` is the same as `[[ -n $var ]]`, which is `true` for any nonempty string. So this check always returns `true`. Which means this function always returns the first origin in the config file. Of course both my manual tests and the unit tests did not catch this as we only have the one remote.

I modified the tests to add a second remote to catch this case, and fixed the check in commit 4c22aa97b89140d1678a4c51546f0640290cf0d7.